sympy/core/assumptions.py

```diff
@@ -515,7 +515,10 @@ def _ask(fact, obj):
 
     # Store None into the assumptions so that recursive attempts at
     # evaluating the same fact don't trigger infinite recursion.
-    assumptions._tell(fact, None)
+    try:
+        assumptions._tell(fact, None)
+    except InconsistentAssumptions:
+        return assumptions[fact]
```

Comment:
There needs to be a comment in the code explaining why this is here and what it does i.e. when would the exception get raised and why is it okay to return this value instead?